File: 0078-subsets/0078-subsets.java
class Solution {
    void Func(int[] nums, List<Integer> temp, List<List<Integer>> res, int start)
    {
        res.add(new ArrayList<>(temp));
        for(int i=start; i<nums.length; i++)
        {
            temp.add(nums[i]);
            Func(nums, temp, res, i+1);
            temp.remove(temp.size() - 1);
        }
    }
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        Func(nums, temp, res, 0);
        return res;
    }
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    void Func(vector<int>& nums, vector<vector<int>>& ans, vector<int> temp, int start){
        ans.push_back(temp);
        for(int i=start; i<nums.size(); i++){
            temp.push_back(nums[i]);
            Func(nums, ans, temp, i+1);
            temp.pop_back();
        }
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> temp;
        Func(nums, ans, temp, 0);
        return ans;
    }
};